spring-cloud-gcp-data-datastore/src/test/java/org/springframework/cloud/gcp/data/datastore/core/convert/DatastoreEntityConverterImplTests.java

```diff
@@ -0,0 +1,167 @@
+import com.google.cloud.datastore.LatLng;
+import com.google.cloud.datastore.NullValue;
+import com.google.cloud.datastore.testing.LocalDatastoreHelper;
+import org.junit.Before;
+import org.junit.Test;
+
+import org.springframework.cloud.gcp.data.datastore.core.mapping.DatastoreDataException;
+import org.springframework.cloud.gcp.data.datastore.core.mapping.DatastoreMappingContext;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * @author Dmitry Solomakha
+ *
+ * @since 1.1
+ */
+
+public class DatastoreEntityConverterImplTests {
+	private static final LocalDatastoreHelper HELPER = LocalDatastoreHelper.create(1.0);
+
+	private Datastore datastore;
+
+	@Before
+	public void setUp() {
+		this.datastore = HELPER.getOptions().toBuilder().setNamespace("ghijklmnop").build().getService();
+	}
+
+	@Test
+	public void readTest() {
+		byte[] bytes = { 1, 2, 3 };
+		Entity entity = Entity.newBuilder(this.datastore.newKeyFactory().setKind("aKind").newKey("1"))
+				.set("stringField", "string value")
+				.set("boolField", true)
+				.set("doubleField", 3.1415D)
+				.set("longField", 123L)
+				.set("latLngField", LatLng.of(10, 20))
+				.set("timestampField", Timestamp.ofTimeSecondsAndNanos(30, 40))
+				.set("blobField", Blob.copyFrom(bytes))
+				.build();
+		DatastoreEntityConverter entityConverter = new DatastoreEntityConverterImpl(new DatastoreMappingContext());
+		TestDatastoreItem item = entityConverter.read(TestDatastoreItem.class, entity);
+
+		assertThat(item.getStringField()).as("validate string field").isEqualTo("string value");
+		assertThat(item.getBoolField()).as("validate boolean field").isTrue();
+		assertThat(item.getDoubleField()).as("validate double field").isEqualTo(3.1415D);
+		assertThat(item.getLongField()).as("validate long field").isEqualTo(123L);
+		assertThat(item.getLatLngField()).as("validate latLng field").isEqualTo(LatLng.of(10, 20));
+		assertThat(item.getTimestampField()).as("validate timestamp field")
+				.isEqualTo(Timestamp.ofTimeSecondsAndNanos(30, 40));
+		assertThat(item.getBlobField()).as("validate blob field").isEqualTo(Blob.copyFrom(bytes));
+	}
+
+	@Test
+	public void readNullTest() {
+		byte[] bytes = { 1, 2, 3 };
+		Entity entity = Entity.newBuilder(this.datastore.newKeyFactory().setKind("aKind").newKey("1"))
+				.set("stringField", new NullValue())
+				.set("boolField", true)
+				.set("doubleField", 3.1415D)
+				.set("longField", 123L)
+				.set("latLngField", LatLng.of(10, 20))
+				.set("timestampField", Timestamp.ofTimeSecondsAndNanos(30, 40))
+				.set("blobField", Blob.copyFrom(bytes))
+				.build();
+		DatastoreEntityConverter entityConverter = new DatastoreEntityConverterImpl(new DatastoreMappingContext());
+		TestDatastoreItem item = entityConverter.read(TestDatastoreItem.class, entity);
+
+		assertThat(item.getStringField()).as("validate null field").isNull();
+	}
+
+	@Test(expected = DatastoreDataException.class)
+	public void readTestException() {
```

Comment:
This could have a more descriptive test name (about specifically what is wrong).